silver/tests/spec/test_payment_method.py

```diff
@@ -1,49 +1,330 @@
+            'customer_pk': self.customer.pk,
+            'payment_method_id': method.pk
+        })
+
+        self.assert_get_data(url, method)
+
+    def test_post_listing(self):
+        processor_url = reverse('payment-processor-detail', kwargs={
+            'processor_name': 'manual'
+        })
+
+        url = reverse('payment-method-list', kwargs={
+            'customer_pk': self.customer.pk
+        })
+
+        response = self.client.post(url, data={
+            'payment_processor': processor_url,
+            'state': PaymentMethod.States.Uninitialized
+        }, format='json')
+
+        method = PaymentMethod.objects.get(customer=self.customer)
+        self.assert_get_data(response.data['url'], method)
+
+    def test_post_listing_additional_data_wrong_state(self):
+        processor_url = reverse('payment-processor-detail', kwargs={
+            'processor_name': 'manual'
+        })
+
+        url = reverse('payment-method-list', kwargs={
+            'customer_pk': self.customer.pk
+        })
+
+        response = self.client.post(url, data={
+            'payment_processor': processor_url,
+            'state': PaymentMethod.States.Uninitialized,
+            'additional_data': '{"random": "value"}'
+        }, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_post_listing_invalid_initial_state(self):
+        invalid_initial_states = list(
+            set([s[0] for s in PaymentMethod.States.Choices]) -
+            {PaymentMethod.States.Uninitialized,
+             PaymentMethod.States.Unverified,
+             PaymentMethod.States.Enabled})
+
+        processor_url = reverse('payment-processor-detail', kwargs={
+            'processor_name': 'manual'
+        })
+
+        url = reverse('payment-method-list', kwargs={
+            'customer_pk': self.customer.pk
+        })
+
+        for state in invalid_initial_states:
+            response = self.client.post(url, data={
+                'payment_processor': processor_url,
+                'state': state
+            }, format='json')
+
+            self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_put_detail_state_transitions(self):
+        states = [x[0] for x in PaymentMethod.States.Choices]
+        permutations = [(old, new) for old in states for new in states]
+
+        valid_transitions = set()
+        for _, transition in iteritems(PaymentMethod.registered_transitions):
+            if isinstance(transition['source'], (tuple, list)):
+                unmerged_transitions = set([(old, transition['target']) for
+                                            old in transition['source']])
+            else:
+                unmerged_transitions = {(transition['source'],
+                                         transition['target'])}
+
+            valid_transitions = valid_transitions.union(unmerged_transitions)
+
+        # add x -> x transactions as they are allowed
```

Comment:
That is what I meant. But in theory nothing stops you from doing it except that we short-circuit the logic if we detect no change, but technically still doable.